packages/block-editor/src/components/block-parent-selector/style.scss

```diff
@@ -9,3 +9,36 @@
 		border-radius: $radius-block-ui;
 	}
 }
+
+// styles for parent block selector when parent block has changes
+.block-editor-block-parent-selector.parent-block-has-changes {
+	left: calc(-#{$grid-unit-60} - #{$grid-unit-30} - #{$border-width});
+
+	// increase the width of parent selector to make room for the dot
+	.block-editor-block-parent-selector__button {
+		width: $grid-unit-60 + $grid-unit-20;
+	}
+
+	// move the parent selector icon to right side to make room for the dot
+	.block-editor-block-icon {
+		margin-right: -15px;
```

Comment:
Changing this value to -16px (or `$grid-unit-20`) seems to get rid of the visual jump for me (tested on a couple of different screens at different viewport sizes).